diff -urwN dflatp-prev/source/dflatp/dfpcomp.h dflatp/source/dflatp/dfpcomp.h
--- dflatp-prev/source/dflatp/dfpcomp.h	2008-03-22 03:00:48.000000000 +0000
+++ dflatp/source/dflatp/dfpcomp.h	2024-03-25 14:00:42.333141251 +0000
@@ -43,8 +43,8 @@
 #include <bios.h>
 #include <ctype.h>
 #include <io.h>
-#include <sys\types.h>
-#include <sys\stat.h>
+#include <sys/types.h>
+#include <sys/stat.h>
 #include <time.h>
 #include <setjmp.h>
 #include "dflatp.h"
@@ -99,7 +99,7 @@
 #define CHECKMARK      (unsigned char) (SCREENHEIGHT==25?251:4)
 #define CASCADEPOINTER (unsigned char) '\x10'
 /* ----------------- title bar characters ----------------- */
-#define CONTROLBOXCHAR (unsigned char) '\xf0'
+#define CONTROLBOXCHAR (unsigned char) '\xfe' //'\xf0' - russian JO
 #define MAXPOINTER     24      /* maximize token            */
 #define MINPOINTER     25      /* minimize token            */
 #define RESTOREPOINTER 18      /* restore token             */
diff -urwN dflatp-prev/source/dflatp/logger.c dflatp/source/dflatp/logger.c
--- dflatp-prev/source/dflatp/logger.c	2008-03-22 10:15:42.000000000 +0000
+++ dflatp/source/dflatp/logger.c	2024-03-25 14:00:42.349141053 +0000
@@ -4,7 +4,7 @@
 //#ifdef DFLATP_LOG
 
 #include <dos.h>
-#include <sys\types.h>
+#include <sys/types.h>
 #include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
diff -urwN dflatp-prev/source/dflatp/menu.h dflatp/source/dflatp/menu.h
--- dflatp-prev/source/dflatp/menu.h	2007-12-26 23:20:18.000000000 +0000
+++ dflatp/source/dflatp/menu.h	2024-03-25 14:01:22.000646415 +0000
@@ -51,4 +51,3 @@
 
 #endif
 
-
diff -urwN dflatp-prev/source/dflatp/portab.h dflatp/source/dflatp/portab.h
--- dflatp-prev/source/dflatp/portab.h	2008-03-22 03:51:04.000000000 +0000
+++ dflatp/source/dflatp/portab.h	2024-03-25 14:00:42.349141053 +0000
@@ -7,6 +7,9 @@
 #include <dos.h>
 
 #if MSC | __WATCOMC__
+#if __WATCOMC__
+#include <stdlib.h>
+#endif
 #include <direct.h>
 #else
 #include <dir.h>
