diff -urN dflatp\source\dflatp\dfpcomp.h newdflatp\source\dflatp\dfpcomp.h
--- dflatp\source\dflatp\dfpcomp.h	Sat Mar 22 03:00:48 2008
+++ newdflatp\source\dflatp\dfpcomp.h	Wed Aug 16 22:22:07 2023
@@ -43,8 +43,8 @@
 #include <bios.h>
 #include <ctype.h>
 #include <io.h>
-#include <sys\types.h>
-#include <sys\stat.h>
+#include <sys/types.h>
+#include <sys/stat.h>
 #include <time.h>
 #include <setjmp.h>
 #include "dflatp.h"
@@ -99,7 +99,7 @@
 #define CHECKMARK      (unsigned char) (SCREENHEIGHT==25?251:4)
 #define CASCADEPOINTER (unsigned char) '\x10'
 /* ----------------- title bar characters ----------------- */
-#define CONTROLBOXCHAR (unsigned char) '\xf0'
+#define CONTROLBOXCHAR (unsigned char) '\xfe' //'\xf0' - russian JO
 #define MAXPOINTER     24      /* maximize token            */
 #define MINPOINTER     25      /* minimize token            */
 #define RESTOREPOINTER 18      /* restore token             */
diff -urN dflatp\source\dflatp\logger.c newdflatp\source\dflatp\logger.c
--- dflatp\source\dflatp\logger.c	Sat Mar 22 10:15:42 2008
+++ newdflatp\source\dflatp\logger.c	Wed Aug 16 21:13:56 2023
@@ -4,7 +4,7 @@
 //#ifdef DFLATP_LOG
 
 #include <dos.h>
-#include <sys\types.h>
+#include <sys/types.h>
 #include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
diff -urN dflatp\source\dflatp\portab.h newdflatp\source\dflatp\portab.h
--- dflatp\source\dflatp\portab.h	Sat Mar 22 03:51:04 2008
+++ newdflatp\source\dflatp\portab.h	Wed Aug 16 21:13:56 2023
@@ -7,6 +7,9 @@
 #include <dos.h>
 
 #if MSC | __WATCOMC__
+#if __WATCOMC__
+#include <stdlib.h>
+#endif
 #include <direct.h>
 #else
 #include <dir.h>
