diff -urwN debug/src/debug.asm debug-new/src/debug.asm
--- debug/src/debug.asm	Mon Aug 19 01:28:07 2024
+++ debug-new/src/debug.asm	Mon Aug 19 01:26:51 2024
@@ -1831,7 +1831,7 @@
 	ret
 setdbgexc0d0e endp
 
-	include <TRAPD.INC>
+	include <trapd.inc>
 
 if CATCHINT21
 intr21pm proc
@@ -8340,14 +8340,14 @@
 
 if RING0
  if LMODE
-	include <TRAPPL.INC>
+	include <trappl.inc>
  elseif V86M
-	include <TRAPPV.INC>
+	include <trappv.inc>
  else
-	include <TRAPP.INC>
+	include <trapp.inc>
  endif
 else
-	include <TRAPR.INC>
+	include <trapr.inc>
 endif
 
 ;--- fall thru
@@ -10259,7 +10259,7 @@
 
 ;--- disassembler code
 
-	include <DISASM.INC>
+	include <disasm.inc>
 
 ;   SHOWMACH - Return strings 
 ;           "[needs _86]" or "[needs _87]",
@@ -10476,7 +10476,7 @@
  if RING0
 	.386
  endif
-	include <FPTOSTR.INC>
+	include <fptostr.inc>
 endif
 
 ;--- the layout for FSAVE/FRSTOR depends on mode and 16/32bit
@@ -10983,9 +10983,9 @@
 	pushcontext cpu
 	.386
  if RING0 and V86M
-	include <DPRINTFR.INC>
+	include <dptrintfr.inc>
  else
-	include <DPRINTF.INC>
+	include <dprintf.inc>
  endif
 	popcontext cpu
 endif
diff -urwN debug/src/disasm.inc debug-new/src/disasm.inc
--- debug/src/disasm.inc	Mon Aug 19 01:28:07 2024
+++ debug-new/src/disasm.inc	Mon Aug 19 01:27:07 2024
@@ -67,7 +67,7 @@
 
 CONST segment
 
-	include <DISTBL.INC>
+	include <distbl.inc>
 
 ;--- table of obsolete-instruction values.
 ;--- instructions are FENI, FDISI, FSETPM, MOV to/from TRx
diff -urwN debug/src/lineasm.inc debug-new/src/lineasm.inc
--- debug/src/lineasm.inc	Mon Aug 19 01:28:07 2024
+++ debug-new/src/lineasm.inc	Mon Aug 19 01:27:21 2024
@@ -150,7 +150,7 @@
 		db 0a6h,0aeh				;REPE/REPNE (CMPSB, SCASB)
 N_REPALL equ $ - replist
 
-	include <ASMTBL.INC>
+	include <asmtbl.inc>
 
 ;--- opindex array - help array to access operand list items (oplists);
 ;--- needed because the operand list items differ in size.
