diff -urwN debug/src/debug.asm debug-new/src/debug.asm
--- debug/src/debug.asm	2024-03-25 17:42:33.606630798 +0000
+++ debug-new/src/debug.asm	2024-03-25 17:41:46.603223510 +0000
@@ -1402,7 +1402,7 @@
 		db 0a6h,0aeh				;REPE/REPNE (CMPSB, SCASB)
 N_REPALL equ $ - replist
 
-	include <DEBUGTBL.INC>
+	include <debugtbl.inc>
 
 opindex label byte
 	.radix 16t
@@ -9810,7 +9810,7 @@
 	include <TRAPP.INC>
  endif
 else
-	include <TRAPR.INC>
+	include <trapr.inc>
 endif
 
 ;--- fall thru
@@ -11521,7 +11521,7 @@
 
 ;--- disassembler code
 
-	include <DISASM.INC>
+	include <disasm.inc>
 
 ;   SHOWMACH - Return strings 
 ;           "[needs _86]" or "[needs _87]",
